scdl/scdl.py

```diff
@@ -381,10 +404,15 @@ def download_playlist(playlist):
         with codecs.open(playlist_name + '.m3u', 'w+', 'utf8') as playlist_file:
             playlist_file.write('#EXTM3U' + os.linesep)
             del playlist['tracks'][:offset - 1]
-            for counter, track_raw in enumerate(playlist['tracks'], offset):
-                logger.debug(track_raw)
-                logger.info('Track n°{0}'.format(counter))
-                download_track(track_raw, playlist['title'], playlist_file)
+            for counter, track_raw in enumerate(sort_tracks(playlist['tracks']), offset):
+                if not arguments['-n'] or counter <= int(arguments['-n']): 
```

Comment:
I see you've added this `counter` variable, but I'm not seeing where it's defined or updated. Sorry if I'm missing something, I'm not super familiar with this app.